includes/admin/class-amp-admin-support.php

```diff
@@ -0,0 +1,721 @@
+<?php
+/**
+ * Class AMP_Admin_Support
+ *
+ * @package AMP
+ * @since 2.1
+ */
+
+use AmpProject\AmpWP\Admin\GoogleFonts;
+use AmpProject\AmpWP\QueryVar;
+
+/**
+ * Class managing support pages in wp-admin dashboard.
+ *
+ * @since 2.1
+ * @internal
+ */
+class AMP_Admin_Support {
+	/**
+	 * Handle for CSS file.
+	 *
+	 * @since 2.1
+	 *
+	 * @var string
+	 */
+	const ASSET_HANDLE = 'amp-settings';
+
+	/**
+	 * Endpoint to send diagnostic data.
+	 *
+	 * @since 2.1
+	 *
+	 * @var string
+	 */
+	const AMP_SEND_DATA_SERVER_ENDPOINT = 'https://insights.amp-wp.org';
+
+	/**
+	 * Registers functionality through WordPress hooks.
+	 *
+	 * @since 2.1
+	 */
+	public function init() {
+		add_action( 'admin_enqueue_scripts', [ $this, 'enqueue_assets' ] );
+		add_action( 'admin_menu', [ $this, 'admin_menu' ] );
+		/**
+		 * AJAX responder.
+		 */
+		add_action( 'wp_ajax_amp_diagnostic', [ $this, 'amp_diagnostic' ] );
+
+		/**
+		 * Add Diagnostic link to Admin Bar.
+		 */
+		add_action( 'admin_bar_menu', [ $this, 'admin_bar_menu' ], 102 );
+
+		/**
+		 * Add diagnostic link to meta box.
+		 */
+		add_filter( 'amp_validated_url_status_actions', [ $this, 'amp_validated_url_status_actions' ], 10, 2 );
+
+		/**
+		 * Add diagnostic link to Post row actions.
+		 */
+		add_filter( 'post_row_actions', [ $this, 'post_row_actions' ], PHP_INT_MAX - 1, 2 );
+
+		/**
+		 * Plugin row Support link.
+		 */
+		add_filter( 'plugin_row_meta', [ $this, 'plugin_row_meta' ], 10, 4 );
+	}
+
+	/**
+	 * Sends diagnostic data to AMP support server.
+	 *
+	 * Prints the UUID returned from the server, if data is sent successfully, else, prints the error message.
+	 *
+	 * @since 2.1
+	 *
+	 * @return void
+	 */
+	public function amp_diagnostic() {
+
+		if (
+			(
+				! current_user_can( 'manage_options' )
+				|| ! check_ajax_referer( 'amp-diagnostic' )
+			)
+			&& ! defined( 'WP_CLI' )
```

Comment:
This is the AJAX callback function. Why are we checking for `WP_CLI` const?

Reply:
Because the AJAX call was originally called by WP-CLI to avoid duplicating code.
Now the code is duplicated, so this line has been removed.